supervisor/README.md

```diff
@@ -0,0 +1,147 @@
+# Cosmos Supervisor
+
+This is a tiny shim around Cosmos SDK binaries that use the upgrade
+module that allows for smooth and configurable management of upgrading
+binaries as a live chain is upgraded, and can be used to simplify validator
+devops while doing upgrades or to make syncing a full node for genesis
+simple. The `supervisor` will monitor the stdout of the daemon to look 
+for messages from the upgrade module indicating a pending or required upgrade 
+and act appropriately. (With better integrations possible in the future).
+
+## Arguments
+
+`supervisor` is a shim around a native binary. All arguments passed to the `supervisor` 
+command will be passed to the current daemon binary (as a subprocess).
+ It will return stdout and stderr of the subprocess as
+it's own. Because of that, it cannot accept any command line arguments, nor
+print anything to output (unless it dies before executing a binary).
+
+Configuration will be passed in the following environmental variables:
+
+* `DAEMON_HOME` is the location where upgrade binaries should be kept (can
+be `$HOME/.gaiad` or `$HOME/.xrnd`)
+* `DAEMON_NAME` is the name of the binary itself (eg. `xrnd`, `gaiad`, `simd`)
+* `DAEMON_ALLOW_DOWNLOAD_BINARIES` (optional) if set to `on` will enable auto-downloading of new binaries
+(for security reasons, this is intended for fullnodes rather than validators)
+* `DAEMON_RESTART_AFTER_UPGRADE` (optional) if set to `on` it will restart the sub-process with the same args
+(but new binary) after a successful upgrade. By default, the `supervisor` dies afterward and allows the supervisor
+to restart it if needed. Note that this will not auto-restart the child if there was an error.
+
+## Folder Layout
+
+`$DAEMON_HOME/upgrade_manager` is expected to belong completely to the supervisor and 
```

Comment:
Shouldn't be change this folder to `$DAEMON_HOME/supervisor` as well?

Reply:
Yeah this is a doubt I have, everyone agree on this change?

Reply:
@alessio @alexanderbez

Reply:
We should all agree on the name to begin with 😉. I was suggesting `cosmos-supervisor` just because that's less generic than `supervisor` but don't have a super strong opinion.

Reply:
Changed!

Reply:
the problem I see with the cosmos-supervisor is that is a little bit long, but I am open to change it, is a fast change.